cocotb/scheduler.py

```diff
@@ -81,6 +83,15 @@ def __exit__(self, *excinfo):
         _profile.disable()
 
 
+@contextmanager
+def task_context(scheduler, task):
```

Comment:
Is there a particular reason this is a free function and not a method on `Scheduler`?

Reply:
I originally tried a custom context manager like `profiling_context` above, but couldn't get it to work correctly, then I found `@contextmanager`. I haven't tried as a method on `Scheduler`.

Reply:
`@contextmanager` works just fine on methods